users/views.py

```diff
@@ -10,3 +12,20 @@ class SignUpView(CreateView):
     form_class = SignUpForm
     template_name = 'users/signup.html'
     success_url = '/'
+
+
+class ProfileView(LoginRequiredMixin, UpdateView):
+    """Page to view the current user profile."""
+    model = User
+    template_name = 'users/profile.html'
+    fields = ['username', 'first_name', 'last_name', 'email', 'avatar']
+    success_url = '/'
+
+    def get_object(self, queryset=None):
+        return self.request.user
+
+    def form_valid(self, form):
+        self.object = form.save(commit=False)
+        self.object.user = self.request.user
+        self.object.save()
+        return HttpResponseRedirect(self.get_success_url())
```

Comment:
Just so you know: django has a `redirect()` shortcut that expects the name of a view (and optional arguments) or an absolute URL:

```python
from django.shortcuts import redirect

# the 'index' view is defined as '/' in uptv/urls.py
return redirect('index')
```

See [docs](https://docs.djangoproject.com/fr/2.1/topics/http/shortcuts/#redirect)